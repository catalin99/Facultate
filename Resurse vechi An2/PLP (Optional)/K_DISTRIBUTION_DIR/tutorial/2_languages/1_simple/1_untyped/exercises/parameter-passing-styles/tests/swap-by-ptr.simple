function swap(x,y) {
  *x = *x + *y;
  *y = *x - *y;
  *x = *x - *y;
}

function main() {
  var x = 7;
  var y = 10;
  print("Before swap: x = ", x, " and y = ", y, "\n");
  swap(&x,&y);
  print("After swap(&x,&y): x = ", x, " and y = ", y, "\n");
  swap(&x,&x);
  print("After swap(&x,&x): x = ", x, " and y = ", y, "\n");
}
