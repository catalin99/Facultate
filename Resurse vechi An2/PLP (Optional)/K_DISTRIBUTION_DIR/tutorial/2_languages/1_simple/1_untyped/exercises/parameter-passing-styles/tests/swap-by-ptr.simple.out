Before swap: x = 7 and y = 10
After swap(&x,&y): x = 10 and y = 7
After swap(&x,&x): x = 0 and y = 7
